src/lib/frontend/translate.ml

```diff
@@ -761,55 +761,139 @@ let handle_patt_var id (DE.{ term_descr; _ } as term)  =
       "Expected a variable in a case match but got %a"
       DE.Term.print term
 
-(** Helper function to translate patterns in a pattern-matching from a Dolmen
-    Term.t to an Alt-Ergo Expr.t *)
-let mk_pattern DE.{ term_descr; _ } =
-  match term_descr with
-  | App (
-      { term_descr =
-          Cst ({ builtin = B.Constructor { adt; case; }; _ } as cst); _
-      }, _, pargs
-    ) ->
-    let vnames =
-      begin match DT.definition adt with
-        | Some (Adt { cases; _ }) ->
-          let { DE.dstrs; _ } = cases.(case) in
-          Array.fold_right (
-            fun v acc ->
-              match v with
-              | Some dstr -> dstr :: acc
-              | _ -> assert false
-          ) dstrs []
-        | _ ->
-          Fmt.failwith
-            "Expected a constructor for an algebraic data type but got\
-             something else for the definition of: %a"
-            DE.Ty.Const.print adt
-      end
+module Match = struct
```

Comment:
Nit: if using local modules for actual implementation work there is an abstraction boundary that should be ideally documented and at least typed.

```suggestion
module Match : sig
  type pat
  val mk_pat : DE.term -> pat
  val make : Expr.t -> (pat * Expr.t) list -> Expr.t
end = struct
```